src/server/visual-diff-reporter.js

```diff
@@ -0,0 +1,101 @@
+import { dirname, join } from 'path';
+import { execSync } from 'child_process';
+import { fileURLToPath } from 'url';
+import { getTestInfo } from './visual-diff-plugin.js';
+import { writeFileSync } from 'fs';
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+function createData(rootDir, sessions) {
+
+	const files = new Map();
+	const browsers = new Map();
+
+	sessions.forEach(s => {
+
+		const browserName = s.browser.name;
+		if (!browsers.has(browserName)) {
+			browsers.set(browserName, {
+				name: browserName,
+				numFailed: 0,
+				version: s.browser.browser.version().substring(0, s.browser.browser.version().indexOf('.'))
+			});
+		}
+		const browserData = browsers.get(browserName);
+
+		const fileName = s.testFile.substring(rootDir.length + 1);
+		if (!files.has(fileName)) {
+			files.set(fileName, { name: fileName, numFailed: 0, tests: new Map() });
+		}
+		const fileData = files.get(fileName);
+		flattenResults(s, browserData, fileData);
+
+	});
+
+	return { files, browsers };
+
+}
+
+function flattenResults(session, browserData, fileData) {
```

Comment:
Test results are nested by "suite", which we don't really care about in the report. This flattens them all (putting " > " in between suites) and converts things into the format that we want.